Makefile

```diff
@@ -81,42 +81,26 @@ test_fast:: build
 	cd pkg && $(GO_TEST_FAST) ${PROJECT_PKGS}
 
 test_build:: $(TEST_ALL_DEPS)
-	cd tests/testprovider && go build -o pulumi-resource-testprovider
-	cd tests/integration/construct_component/testcomponent && yarn install && yarn link @pulumi/pulumi && yarn run tsc
-	cd tests/integration/construct_component/testcomponent-go && go build -o pulumi-resource-testcomponent
-	cd tests/integration/construct_component/testcomponent-python && $(PYTHON) -m venv venv && venv/bin/python -m pip install -e ../../../../sdk/python/env/src
-	cd tests/integration/construct_component_output_values/testcomponent && yarn install && yarn link @pulumi/pulumi && yarn run tsc
-	cd tests/integration/construct_component_output_values/testcomponent-go && go build -o pulumi-resource-testcomponent
-	cd tests/integration/construct_component_output_values/testcomponent-python && $(PYTHON) -m venv venv && venv/bin/python -m pip install -e ../../../../sdk/python/env/src
+	cd tests/testprovider && go build -o pulumi-resource-testprovider$(shell go env GOEXE)
+	PYTHON=$(PYTHON) ./scripts/prepare-test.sh construct_component
```

Comment:
Thanks for cleaning this up!

Reply:
Happy to, I was forced to because cross-platform scripting requires sourcing `activate` and doing it in Make that runs one process per line was getting very, very unreadable.